index.js

```diff
@@ -0,0 +1,23 @@
+import fp from 'fastify-plugin'
+
+function fastifyConstraints(fastify, options, next) {
+  fastify.addHook('onRegister', (instance, opts) => {
+    if (!opts?.constraints) {
+      next()
+      return
+    }
+
+    instance.addHook('onRoute', function hook(routeOptions) {
```

Comment:
Not sure if it is good or not. It may add a lot of `onRoute` function.

Reply:
I see your point, we would have as many on route hooks as plugins with constraints. I think your way would be more efficient. Let me change that.